### ISSUE TITLE
Should disable `unicorn/require-array-sort-compare` when `@typescript-eslint/require-array-sort-compare` is used

### ISSUE THREAD

**Issue by andersk** [xojs/eslint-config-xo#104]
In a TypeScript file:

```ts
[8, 9, 10].sort();
['a', 'b', 'c'].sort();
```

```
  test.ts
  ✖  1:1   Require compare argument.                                      @typescript-eslint/require-array-sort-compare
  ✖  1:12  Pass a compare function to avoid sorting elements as strings.  unicorn/require-array-sort-compare
  ✖  2:17  Pass a compare function to avoid sorting elements as strings.  unicorn/require-array-sort-compare

  3 errors
```

The first two errors are duplicates. The third is a false positive, since sorting strings as strings is fine.

We should disable `unicorn/require-array-sort-compare` in TypeScript in favor of `@typescript-eslint/require-array-sort-compare`, as its [documentation suggests](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/require-array-sort-compare.md).
